sqlglot/dialects/redshift.py

```diff
@@ -434,3 +440,7 @@ def array_sql(self, expression: exp.Array) -> str:
                 return super().array_sql(expression)
 
             return rename_func("ARRAY")(self, expression)
+
+        def explode_sql(self, expression: exp.Explode) -> str:
+            self.unsupported("Unsupported EXPLODE() function")
+            return ""
```

Comment:
I'm not sure about this, e.g. you can have `Explode` under `FROM` and `JOIN` in Spark. Wouldn't we be able to transpile these to Redshift? What do we do today?

Reply:
We don't do anything, it will get transpiled as is to Redshift. Maybe we can point Redshift's `exp.Explode` to it's `unnest_sql()`?

Reply:
Sounds reasonable 👍

Reply:
After some thought I remembered why I thought I'd put an unsupported warning there, the "implicit unnest" syntax in Redshift is the following:

```SQL
with t as (select array(1, 2) as foo) select foo2 from t as tbl, tbl.foo as foo2

1
2
```

As you said, Spark's `EXPLODE()` can also be used in FROM, e.g:

```
spark-sql (default)> select * from explode(array(1, 2)); 
column
1
2
```

Or as a lateral view:
```
spark-sql (default)> with t as (select array(1) as foo) select * from t lateral view explode(foo) as col;
foo     col
[1]     1
```

However, by transforming `exp.Explode` to an `exp.Unnest` @ generation time we end up with the "implicit unnest" syntax (i.e just dropping the surrounding `UNNEST(...)`), which is not actually unnesting the array:

```
select * from array(1, 2)
[1, 2]
```

So, the semantics do not seem to match well here for the transformation to make much sense. Wdyt?

Reply:
This doesn't look that much different from what we do for `Unnest` today, though:

```
(.venv) ➜  sqlmesh git:(main) duckdb
D SELECT * FROM UNNEST([1, 2, 3]);
┌──────────────────────────┐
│ main.list_value(1, 2, 3) │
│          int32           │
├──────────────────────────┤
│                        1 │
│                        2 │
│                        3 │
└──────────────────────────┘
D

(.venv) ➜  sqlmesh git:(main) python
Python 3.12.5 (main, Aug  6 2024, 19:08:49) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlglot
>>> sqlglot.transpile("SELECT * FROM UNNEST([1, 2, 3]);", "duckdb", "redshift")
['SELECT * FROM ARRAY(1, 2, 3)']
```

So, generating `Explode` using `unnest_sql` should _at least_ work for the same cases as `Unnest` does, right? For example what if you had lateral / cross joins, are those transpiled correctly?

Reply:
I guess the main case we aim to solve for is something like:

```
with t as (select [1, 2, 3] as foo) select * from t, UNNEST(t.foo);
┌───────────┬────────┐
│    foo    │ unnest │
│  int32[]  │ int32  │
├───────────┼────────┤
│ [1, 2, 3] │      1 │
│ [1, 2, 3] │      2 │
│ [1, 2, 3] │      3 │
└───────────┴────────┘
```

This currently is transpiled to the invalid Redshift:
```
WITH t AS (SELECT ARRAY(1, 2, 3) AS foo) SELECT * FROM t, t.foo
```


To make it work, the transformation requires a couple tweaks regarding the aliases which are not done yet:

```
WITH t AS (SELECT ARRAY(1, 2, 3) AS foo) SELECT * FROM t as tbl, tbl.foo as foo2;

[1, 2, 3]    1
[1, 2, 3]    2
[1, 2, 3]    3
```

I'm not sure if there's a valid way to transpile `SELECT * FROM UNNEST/EXPLODE`, maybe we'd have to generate our own aliases and change the projection list (?)

Reply:
Ok sg, no worries.